src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -1,92 +1,123 @@
-/* Copyright (C) 2016 NooBaa */
+/* Copyright (C) 2022 NooBaa */
 'use strict';
 
 const _ = require('lodash');
 const { v4: uuid } = require('uuid');
 const dbg = require('../../../util/debug_module')(__filename);
 const S3Error = require('../s3_errors').S3Error;
 
+// parse lifecycle rule filter
+function parse_filter(filter) {
+    const current_rule_filter = {};
+    if (filter.Tag && filter.Tag.length === 1) {
+        const tag = filter.Tag[0];
+        current_rule_filter.tags = [{ key: tag.Key[0], value: tag.Value[0] }];
+    }
+    if (filter.Prefix && filter.Prefix.length === 1) {
+        current_rule_filter.prefix = filter.Prefix[0];
+    }
+    if (filter.ObjectSizeGreaterThan &&
```

Comment:
- Can we set ObjectSizeGreaterThan and ObjectSizeLessThan that is an empty group in s3?

Reply:
It's a good idea to verify invalid size range